pandas/core/window/common.py

```diff
@@ -109,6 +109,9 @@ def dataframe_from_int_dict(data, frame_template):
 
                     # set the index and reorder
                     if arg2.columns.nlevels > 1:
+                        # mypy needs to know columns is a MultiIndex, Index doesn't
+                        # have levels attribute
+                        assert isinstance(arg2.columns, MultiIndex)
```

Comment:
the nlevels check guarantees a MultiIndex, so could cast. (and can keep the comment as is)

Reply:
Done.

Reply:
out of curiosity i did some timings:

```
In [2]: dti = pd.date_range("2016-01-01", periods=3)

In [3]: tdi = dti- dti[0]

In [4]: mi = pd.MultiIndex.from_product([dti, tdi])

In [10]: %timeit assert isinstance(mi, pd.MultiIndex)
94.5 ns ± 1.43 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [11]: %timeit mi2 = cast("MultiIndex", mi)
95.8 ns ± 0.332 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [14]: %timeit assert isinstance(mi, MultiIndex)
65.7 ns ± 1.25 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

Not big enough a difference to matter, but i had expected the `cast` to be faster.

Reply:
I agree.

I issue I have with many isinstance checks is that they overlap/duplicate the is_* check (or in this case, the ndim check) used that mypy can't narrow the types from.

Once we add an runtime instance check we might as well change the condition instead.